crates/autopilot/src/run_loop.rs

```diff
@@ -158,18 +160,20 @@ impl RunLoop {
         if let Some(Participant { driver, solution }) = solutions.last() {
             tracing::info!(driver = %driver.name, solution = %solution.id(), "winner");
 
+            let start = Instant::now();
             let revealed = match self.reveal(driver, auction_id, solution.id()).await {
                 Ok(result) => {
-                    Metrics::reveal_ok(driver);
+                    Metrics::reveal_ok(driver, start.elapsed());
                     result
                 }
                 Err(err) => {
-                    Metrics::reveal_err(driver, &err);
+                    Metrics::reveal_err(driver, start.elapsed(), &err);
                     tracing::warn!(driver = %driver.name, ?err, "failed to reveal winning solution");
                     return;
                 }
             };
 
+            let start = Instant::now();
```

Comment:
maybe give those slightly more descriptive names (to avoid accidental reuse of a different `start` variable)